File: 2_semester/Laba_3/22/3.py
def func(number_f, power_f, divisor_f):
    number_f= number_f % divisor_f
    remainder = 1
    for i in range (1, power_f+1):
        remainder*=number_f
        remainder= remainder % divisor_f
    return remainder

print("Enter a number and a power of the number: ")
number, power =map(int, input().split())
print("Enter a divisor: ")
divisor = int(input())
print("Remainder is", func(number, power, divisor))

File: 2_semester/Laba_3/22/3.cpp
#include <iostream>
using namespace std;

int func(int number, int power, int div)
{
    power = power % (div-1);
    number= number % div;
    int res = 1;
    for (int x =1; x<=power; x++)
    {
        res*=number;
        res= res % div;
    }
    return res;
}

int main()
{
    int number, power, div;
    cout << "Enter a number and a power of the number: ";
    cin >> number >> power;
    cout << "Enter a div: ";
    cin >> div;
    cout << "res is " <<  func(number, power, div);
}